Aiguzhin_V-task02 (ver.1)

Bushpulbek opened the pull request:
Алексей Алексеевич, здравствуйте!

Решил попробовать сделать не в виде списка списков, а с помощью словарей.
P.S. никогда так больше делать не буду

Как итог, выводит лишь до 2 уровня вложенности:
  1. Пока не очень понимаю, как создать нормальную рекурсию через fill_tree();
  2. Возможно, сама идея использовать словари неудобна для данной задачи;
  3. Может хорошим решением были бы классы, но изучать пока их лень)

Bushpulbek committed:
Aiguzhin_V-task02 (ver.1)

xthebat commented on Aiguzhin_V/task02/main.py:
```diff
@@ -0,0 +1,88 @@
+    if len(args) > 1:
+        sys.exit("There must be only one argument - path of root directory")
+    # Check the validity of an argument
+    elif not isinstance(args[0], str):
```
Если это аргумента командной строки - то эта проверка лишняя, там не может быть не строка. Если проверка сделана для тестов - то тут уже этой частью функциональности будет пользоваться программист, которому проще по исключению и stacktrace найти в чем проблема

xthebat commented on Aiguzhin_V/task02/main.py:
```diff
@@ -0,0 +1,88 @@
+    # If above checks are passed
+    else:
+        # Return the name of the current directory
+        if args[0] in current_dir_signs:
```
Скорее здесь проверить на кол-во аргументов, если их ноль - то тогда использовать текущую директорию. Кроме этого кодирование в виде пробела текущей директории никогда не видел. 
Лучше использовать os.path.realpath() - чтобы развернуть "." в абсолютный путь

xthebat commented on Aiguzhin_V/task02/main.py:
```diff
@@ -0,0 +1,88 @@
+
+
+# Example of compiling structure:
+# {'dir1': [{'dir11': ['dir11_1', 'dir11_2']}, 'dir12', {'dir13': 'dir13_1'}, {'dir14': [{'dir15_2': 'dir15_2-1'}]}]}
```
Структура dict для хранения дерева - очень хорошо подходит
```
{
   "dir1:" {
      "dir11": {
          "file1": None
          "file2": None
       }
       "dir12": {
          "file1": None
          "file2": None
       }
   }
 "dir2:" {
      "dir21": {
          "file1": None
          "file2": None
       }
       "dir22": {
          "file1": None
          "file2": None
       }
   } 
}
```

xthebat replied:
Та что получилась структура - довольно сложная, если использовать словарь, то cписки вообще не нужны, все можно сделать на одних словарях, см. выше

xthebat commented on Aiguzhin_V/task02/main.py:
```diff
@@ -0,0 +1,88 @@
+    if nested_files:
+        dirs_in_nest = []
+        # Check if received values are directories
+        is_dir_values = list(map(os.path.isdir, [given_root + "\\" + file_name for file_name in nested_files]))
```
Не очень понял зачем в начале через map искать директории, тогда уже лучше filter, и потом все равно эту проверку можно было вставить в for вместо is_dir

xthebat commented on Aiguzhin_V/task02/main.py:
```diff
@@ -0,0 +1,88 @@
+
+
+# Returns list of nested first-level directories of given root
+def get_nested_dirs(given_root: str) -> list:
```
Зачем get_nested_dirs отделять от fill_tree - по факту тут кода минимум - реально это только проверка os.path.isdir

xthebat commented on Aiguzhin_V/task02/main.py:
```diff
@@ -0,0 +1,88 @@
+def main(args: list) -> None:
+    start_point = arg_check(args)
+    start_depth = 0
+    full_path = dict()
```
Это можно вынести в аргумент по умолчанию

xthebat reviewed (changes requested):
Score: 1/4

На такие аргументы пустой вывод:
<img width="915" alt="image" src="https://user-images.githubusercontent.com/2856140/156547811-f5361c0f-0377-417b-ac67-d5e33e82f055.png">

Bushpulbek committed:
Здравствуйте, Алексей Алексеевич! Спасибо большое за комментарии. Задание сделать получилось (вроде)))

xthebat reviewed (approved):
Score: 4/4 + 1

Commit:
Merge branch 'task02' into task02
